hail/src/main/java/is/hail/io/compress/BGzipInputStream.java

```diff
@@ -202,9 +213,6 @@ public int read(byte[] b, int off, int len) throws IOException {
         System.arraycopy(outputBuffer, outputBufferPos, b, off, toCopy);
         outputBufferPos += toCopy;
 
-        if (outputBufferPos == outputBufferSize)
```

Comment:
I think if you seek exactly to the end of a block, the position will be (o_b, n_b) instead of (o_{b+1}, 0).

Reply:
We should have a test for this (seek to start or middle of a block, then read to the end).